### PR TITLE
Lazy initialize HttpRouteStatsTracker in MethodHandlers

### PR DESCRIPTION
**Pull request by original-brownbear**
Random find from profiling/heap-dump-analysis.
We use about 1M for the route stats trackers instances per ES instance. Making this lazy init should come at a trivial overhead, save most of that 1M (since most nodes only see a tiny fraction of all APIs ever called, at least that's what production heap dumps suggest) and in fact makes the computation of the node stats cheaper by saving spurious sums on 0-valued long adders.

### CODE REVIEW

**Review by rjernst — COMMENTED** [review]
The change looks ok, but it seems a little clunky. I wonder if we could hide the fact this is lazy initialized a little more. With this change there are now 2 methods that must be understood, getStats, and getOrInitStats. Could we have a readonly tracker that has all zeros, and return that in the null case? Then we still need just one method. Alternatively, keep the new stats tracker internal, and combine the add methods into a single update method?

**Comment by original-brownbear** [comment]
Sorry @rjernst this fell off my radar. 

> Could we have a readonly tracker that has all zeros

How about https://github.com/elastic/elasticsearch/pull/114107/commits/0d16de22f0eb72a3f34ceb42c7fd4322c2b97ec3 for that? It's hard to have a single stats update method because we have the two possible paths for chunked and not-chunked responses. But It was indeed somewhat unnecessary to have two methods for getting the tracker. Getting the stats can be its own thing that returns a constant and getting the tracker can transparently init the tracker instance if needed. WDYT? :)

File: server/src/main/java/org/elasticsearch/rest/RestController.java [hunk 1]
```diff
@@ -879,7 +880,7 @@ public void sendResponse(RestResponse response) {
     private static final class ResourceHandlingHttpChannel extends DelegatingRestChannel {
         private final CircuitBreakerService circuitBreakerService;
         private final int contentLength;
-        private final MethodHandlers methodHandlers;
+        private final HttpRouteStatsTracker stats;
```
Comment (rjernst) [inline server/src/main/java/org/elasticsearch/rest/RestController.java:883]: nit: maybe `statsTracker`?

**Review by rjernst — APPROVED** [review]
LGTM

**Comment by original-brownbear** [comment]
Thanks Ryan!

**Pull request merged**